[WIP] Remove Swift 3 @objc inference support.

rudkx opened the pull request:
This does not include any test updates. I will open a separate PR for them and merge that PR before this one.

Commit:
Remove Swift 3 @objc inference support.

rudkx commented on lib/Sema/TypeCheckDeclOverride.cpp:
```diff
@@ -1203,39 +1203,7 @@ namespace  {
-      Diags.diagnose(Base, diag::make_decl_objc, Base->getDescriptiveKind())
-        .fixItInsert(Base->getAttributeInsertionLoc(false),
-                     "@objc ");
+      // FIXME: Removing this results in compile-time errors.
```
@DougGregor I'm apparently missing what meta-programming magic is requiring an implementation of visitObjCAttr here.

DougGregor replied:
You can use `UNINTERESTING_ATTR(ObjC)` in place of an empty function body.

Commit:
Minor cleanups.

Use UNINTERESTING_ATTR and remove empty function body.

Remove a couple unused variables.

Commit:
Merge branch 'master' into remove-swift3-objc-inference

rudkx commented:
@swift-ci Please smoke test

rudkx commented:
This has a bunch of conflicts at this point; closing.
